data/list/basic.lean

```diff
@@ -795,6 +795,27 @@ theorem update_nth_eq_take_cons_drop (a : α) {n l} (h : n < length l) :
   update_nth l n a = take n l ++ a :: drop (n+1) l :=
 by rw [update_nth_eq_modify_nth, modify_nth_eq_take_cons_drop _ h]
 
+@[simp] lemma update_nth_eq_nil {β} (l : list β) (n : ℕ) (b : β) : l.update_nth n b = [] ↔ l = [] :=
```

Comment:
This proof can be further simplified. The idea is to follow the cases of the definition of `update_nth` (we don't need induction itself). 

So the proof is:
```lean
@[simp] lemma update_nth_eq_nil {β} : ∀(l : list β) (n : ℕ) (b : β), l.update_nth n b = [] ↔ l = []
| []       n       b  := iff.refl _
| (b :: l) 0       b' := by simp [update_nth]
| (b :: l) (n + 1) b' := by simp [update_nth]
```

short and concise. And by using the equation compiler one still clearly sees what is happening.